experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -49,4 +76,33 @@ export async function useAzureSocketIO(
   return this;
 }
 
+function getDefaultNegotiateHandler(
+  engine: WebPubSubEioServer
+): (
+  req: IncomingMessage,
+  res: ServerResponse,
+  getClientAccessTokenUrl: (options?: GenerateClientTokenOptions) => Promise<string>
+) => Promise<void> {
+  return async (
+    req: IncomingMessage,
+    res: ServerResponse,
+    getClientAccessTokenUrl: (options?: GenerateClientTokenOptions) => Promise<string>
+  ): Promise<void> => {
+    let statusCode = 400,
+      message = "Bad Request";
+    try {
+      const username = parse(req.url || "", true).query["username"] as string;
```

Comment:
we shouldn't read query `username`, it is just for simple demo purpose, follow the same logic how express handles auth middleware and read the user info

Reply:
Now we require users provide a method to parse `userId` and expirationn time from `req` if they want to use negotiate